pkg/storage/internalstorage/util.go

```diff
@@ -19,9 +20,27 @@ import (
 const (
 	SearchLabelFuzzyName = "internalstorage.clusterpedia.io/fuzzy-name"
 
-	URLQueryWhereSQL = "whereSQL"
+	URLQueryWhereSQL           = "whereSQL"
+	URLQueryWhereSQLParameters = "whereSQLParameters"
 )
 
+func applyListOptionsURLQueryToWhereQuery(query *gorm.DB, urlQuery url.Values) *gorm.DB {
+	whereClause, ok := urlQuery[URLQueryWhereSQL]
+	if !ok || len(whereClause) == 0 {
+		return query
+	}
+
+	// If a string of numbers is passed in from SQL, the query will be taken as ID by default.
+	// If the SQL contains English letter, it will be passed in as column.
+
+	whereParameters, ok := urlQuery[URLQueryWhereSQLParameters]
+	if !ok || len(whereParameters) == 0 {
+		return query.Where(whereClause[0])
+	}
+
+	return query.Where(whereClause[0], whereParameters)
```

Comment:
need to unpack the params
```go
	return query.Where(whereClause[0], whereParameters...)
```

Reply:
In addition, we need to be able to support the case where the parameters contain arrays, such as a sql statement that contains `IN`, for example `cluster=? AND namespace IN ?`